### ISSUE TITLE
Add `upgrade` option to `flit install`

### ISSUE THREAD

**Issue by mgax** [pypa/flit#681]
When developing a library that is set up with Flit, it's common to run `flit install` to try out the library in a virtualenv. After a while, dependencies might get out of date, so one might want to upgrade them, within the constraints of the library's dependencies. Right now it seems that you'd have to recreate the virtualenv and re-run `flit install` to achieve this.

**Comment by takluyver** [comment]
I think `pip install . --upgrade-strategy eager` would do it. You can also pass that through flit as an environment variable: `PIP_UPGRADE_STRATEGY=eager flit install`.

I'm reluctant to add wrappers for all of pip's many CLI options. `flit install` is kind of only hanging around because I prefer my `--symlink` option to the way editable installs work with pip. :shrug:

**Comment by mgax** [comment]
> I think `pip install . --upgrade-strategy eager` would do it.

Nope, but `pip install . --upgrade-strategy eager --upgrade` will upgrade dependencies.

> You can also pass that through flit as an environment variable: `PIP_UPGRADE_STRATEGY=eager flit install`.

That also doesn't upgrade things, perhaps for the same reason.

---
@takluyver thanks for looking into this! TBH I don't really mind using `pip install -e . --upgrade`, this just seemed like an obvious missing feature of `flit install`.
